src/poliastro/twobody/orbit.py

```diff
@@ -610,6 +615,155 @@ def parabolic(
+
+        Assuming that e is negligible compared to J2, it can be shown that:
+
+        .. math::
+            \overline {e}\approx -\dfrac {J_{3}R_{E}}{2J_{2}\overline {a}}\sin \overline {i}
+
+        The implementation is divided in the following cases:
+
+        1- When the user gives a negative altitude, the method will raise a ValueError
+        2- When the attractor has not defined J2 or J3, the method will raise an AttributeError
+        3- When the attractor has J2/J3 outside of range 1 to 10 , the method will raise an NotImplementedError. Special case for Venus . See "Extension of the critical inclination" by Xiaodong Liu, Hexi Baoyin, and Xingrui Ma
+        4- When argp and inc are both given but neither are critical values, the method will raise a ValueError
+        5- If argp is not given or the given argp is a critical value, the last formula is used to approximate eccentricity
+        6- If inc is not given or the given inc is critical, we set the eccentricity value the same as the Earth's moon.
+
+
+        Parameters
+        ----------
+        attractor : Body
+            Main attractor.
+        alt : ~astropy.units.Quantity
+            Altitude over surface.
+        inc : ~astropy.units.Quantity, optional
+            Inclination, default to critical value.
+        argp : ~astropy.units.Quantity, optional
+            Argument of the pericenter, default to critical value.
+        raan : ~astropy.units.Quantity, optional
+            Right ascension of the ascending node, default to 0 deg.
+        arglat : ~astropy.units.Quantity, optional
+            Argument of latitude, default to 0 deg.
+        epoch: ~astropy.time.Time, optional
+            Epoch, default to J2000.
+        plane : ~poliastro.frames.Planes
+            Fundamental plane of the frame.
+        """
+
+        try:
+            cls._frozen_check_preconditions(attractor, alt)
+
+            argp, inc = cls._frozen_load_defaults_if_needed(attractor, argp, inc)
+            a = attractor.R + alt
+
+            critical_argp = cls._find_closest_value(argp, attractor.critical_argps)
+            if np.isclose(argp, critical_argp, 1e-8, 1e-5 * u.rad):
+                return cls._frozen_critical_argp(
+                    attractor, a, inc, argp, raan, arglat, epoch, plane
+                )
+
+            critical_inclination = cls._find_closest_value(
+                inc, attractor.critical_inclinations
+            )
+            if np.isclose(inc, critical_inclination, 1e-8, 1e-5 * u.rad):
+                return cls._frozen_critical_inclination(
+                    attractor, a, inc, argp, raan, arglat, epoch, plane
+                )
+
+            raise ValueError("Can't create a frozen orbit from given arguments")
+
+        except Exception as exception:
+            cls._frozen_handle_exception(exception, attractor)
+
+    @classmethod
+    def _find_closest_value(cls, value, values):
```

Comment:
This function is generic enough that can go to `poliastro.util`, maybe? There's nothing specific to critical inclinations or even orbits here.